include/boost/fusion/container/generation/make_vector.hpp

```diff
@@ -10,7 +10,59 @@
 #include <boost/fusion/support/config.hpp>
 #include <boost/fusion/container/vector/vector.hpp>
 
+#if !defined(BOOST_FUSION_HAS_VARIADIC_VECTOR)
 # include <boost/fusion/container/generation/detail/pp_make_vector.hpp>
+#else
 
+///////////////////////////////////////////////////////////////////////////////
+// C++11 variadic interface
+///////////////////////////////////////////////////////////////////////////////
+
+#include <boost/fusion/support/detail/as_fusion_element.hpp>
+
+namespace boost { namespace fusion
+{
+    namespace result_of
+    {
+        template <typename ...T>
+        struct make_vector
+        {
+            // make `make_vector<T..., void_...>' into `make_vector<T...>'
+            template <typename, typename...> struct trim_void;
```

Comment:
@djowel said previously:
> The use of fusion::_void is undocumented. It is OK to stop supporting trailing _void if we make sure all uses are fixed in spirit (or elsewhere).

If something in spirit needs to be done to remove this compatibility mode, I can do it if you lack the time. My thinking is that this compatibility mode requires recursive instantations, and it will be less work at compile time if this function is left to a simple variadic expansion like the `make_list` code above. This is probably something for a future commit though - do you know of other Boost projects that fail if this logic isn't performed?